### PR TITLE
Remove old Docker infrastructure

### PR DESCRIPTION
**Pull request by tamird**
@tschottdorf @petermattis

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -108,35 +108,6 @@ Now let's talk to this node. The easiest way to do that is to use the `cockroach
 
 Check out `./cockroach help` to see all available commands.
 
-#### Building the Docker images yourself
-See [build/README.md](build/) for more information on the available Docker
-images `cockroachdb/cockroach` and `cockroachdb/cockroach-dev`.
-You can build both of these images yourself:
-
-* `cockroachdb/cockroach-dev`: `(cd build ; ./build-docker-dev.sh)`
-* `cockroachdb/cockroach`: `(cd build ; ./build-docker-deploy.sh)`
```
Comment (petermattis) [inline README.md]: `build-docker-deploy.sh` still exists. Should this info be kept around?

**Comment by petermattis** [comment]
LGTM on the file deletions. Wondering if we should keep around some of the docker stuff in `README.md` that still applies.

File: README.md (hunk 1 above)
Comment (tamird) [inline README.md]: yeah, it exists, but I don't think we still want to point users at it; we've had some newcomers get tripped up on this docker stuff, so i think having folks treat cockroach like any other go project is going to be a nice simplification

Comment (tschottdorf) [inline README.md]: I don't think building the deploy image is something we need to talk about in `README.md`.

**Comment by tschottdorf** [comment]
LGTM. There are corresponding repositories on the docker registry we could delete as well.

File: README.md (hunk 1 above)
Comment (petermattis) [inline README.md]: Fair enough. Delete away.

**Comment by tamird** [comment]
After merging this, I'm going to delete:
https://hub.docker.com/r/cockroachdb/cockroach-dev/
https://hub.docker.com/r/cockroachdb/dnsmasq/
https://github.com/cockroachdb/docker-builds

**Comment by tschottdorf** [comment]
:+1:

I think we can also kill `clog`, with the caveat that @MBerhault (maybe) still uses it in `cockroach-prod`.

**Comment by MBerhault** [comment]
@tschottdorf is there another clog than util/log/clog.go? If so, I'm not sure why this would be impacted by this change.
But either way, there's no reference to clog in cockroach-prod.

**Comment by tamird** [comment]
removed the clog repo

**Comment by tschottdorf** [comment]
@MBerhault, sorry, that came from memory. I thought I remembered seeing it somewhere. If that's not the case, let's raze that one too.

**Pull request merged**